plugin/pkg/watch/watch.go

```diff
@@ -0,0 +1,12 @@
+package watch
+
+// Chan is used to inform the server of a change
+type Chan chan []string
+
+// Watchable is the interface watchable plugins should implement
+type Watchable interface {
+	Name() string
+	SetWatchChan(Chan)
+	Watch(qname string) error
```

Comment:
is qname fully qualified or not? You probably want to document each method in the interface to help implementers.

Reply:
does this inherit downwards, i.e. is qname a domain and will I get updates for everything under it? I guess no, but docs should make this clearer

Reply:
exact name only if I remember right. and yes, FQDN.

Reply:
done